src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PensionAnnexDateStampServiceTest.java

```diff
@@ -0,0 +1,109 @@
+
+import org.apache.pdfbox.Loader;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.interactive.form.PDAcroForm;
+import org.apache.pdfbox.pdmodel.interactive.form.PDField;
+import org.apache.pdfbox.pdmodel.interactive.form.PDTextField;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseDocument;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.document.Document;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.evidencemanagement.EvidenceManagementDownloadService;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.evidencemanagement.EvidenceManagementUploadService;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.TestConstants.AUTH_TOKEN;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.util.TestResource.fileUploadResponse;
+
+@ExtendWith(MockitoExtension.class)
+class PensionAnnexDateStampServiceTest {
+
+    @InjectMocks
+    private PensionAnnexDateStampService service;
+    @Mock
+    private EvidenceManagementUploadService emUploadService;
+    @Mock
+    private EvidenceManagementDownloadService emDownloadService;
+    @Mock
+    private GenericDocumentService genericDocumentService;
+    private final String caseId = "123123123";
+    private final LocalDate approvalDate = LocalDate.of(2024, 12, 31);
+    @Captor
+    private ArgumentCaptor<List<MultipartFile>> filesCaptor;
+    private CaseDocument document;
+
+    @BeforeEach
+    void setUp() {
+        document = CaseDocument.builder()
+            .documentBinaryUrl("https:mockurl/binary")
+            .documentFilename("Testfile")
+            .documentUrl("http:mockfile").build();
+    }
+
+    @Test
+    void shouldAddApprovalDateToPensionOrderDocument() throws Exception {
+        byte[] docInBytes = loadResource("/fixtures/P1_pension_sharing_annex.pdf");
+        when(emDownloadService.download(document.getDocumentBinaryUrl(), AUTH_TOKEN))
+            .thenReturn(docInBytes);
+        when(emUploadService.upload(any(), anyString(), any()))
+            .thenReturn(fileUploadResponse());
+        when(genericDocumentService.toCaseDocument(any(Document.class))).thenCallRealMethod();
```

Comment:
Interesting.  Is thenCallRealMethod needed?  Looks as though you could leave this line out and it should still work?

Reply:
I think this approach makes sense. The method toCaseDocument doesn't use any of the GenericDocumentService dependencies

Reply:
Right! I didn't consider from that perspective.  Thanks.